rsocket/RSocketClient.cpp

```diff
@@ -6,104 +6,147 @@
 #include "rsocket/RSocketStats.h"
 #include "rsocket/framing/FrameTransport.h"
 #include "rsocket/framing/FramedDuplexConnection.h"
+#include "rsocket/internal/ClientResumeStatusCallback.h"
 #include "rsocket/internal/FollyKeepaliveTimer.h"
 #include "rsocket/internal/RSocketConnectionManager.h"
 
 using namespace folly;
 
 namespace rsocket {
 
-RSocketClient::RSocketClient(
-    std::unique_ptr<ConnectionFactory> connectionFactory)
-    : connectionFactory_(std::move(connectionFactory)),
-      connectionManager_(std::make_unique<RSocketConnectionManager>()) {
-  VLOG(1) << "Constructing RSocketClient";
+RSocketClient::~RSocketClient() {
+  VLOG(4) << "RSocketClient destroyed ..";
 }
 
-RSocketClient::~RSocketClient() {
-  VLOG(1) << "Destroying RSocketClient";
+std::shared_ptr<RSocketRequester> RSocketClient::getRequester() const {
+  return requester_;
 }
 
-folly::Future<std::unique_ptr<RSocketRequester>> RSocketClient::connect(
+RSocketClient::RSocketClient(
+    std::unique_ptr<ConnectionFactory> connectionFactory,
     SetupParameters setupParameters,
     std::shared_ptr<RSocketResponder> responder,
     std::unique_ptr<KeepaliveTimer> keepaliveTimer,
     std::shared_ptr<RSocketStats> stats,
-    std::shared_ptr<RSocketNetworkStats> networkStats) {
+    std::shared_ptr<RSocketNetworkStats> networkStats,
+    std::shared_ptr<ResumeManager> resumeManager,
+    std::shared_ptr<ColdResumeHandler> coldResumeHandler,
+    OnRSocketResume)
+    : connectionFactory_(std::move(connectionFactory)),
+      connectionManager_(std::make_unique<RSocketConnectionManager>()),
```

Comment:
ConnectionManager has a nice way of closing the RSocket connection and cleaning the state.  Just reusing its logic here, instead of replicating the code.